core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -284,13 +309,12 @@ public void start(Listener<RespT> observer, Metadata.Headers headers) {
         transport = obtainActiveTransport();
       } catch (RuntimeException ex) {
         stream = new NoopClientStream();
-        listener.closed(Status.INTERNAL.withDescription("Failed starting transport").withCause(ex),
-            new Metadata.Trailers());
+        listener.closed(Status.fromThrowable(ex), new Metadata.Trailers());
```

Comment:
I just considered several new ways of having the description. I'm really thinking right now trying to supplement the `Status` doesn't add much. Even if it is a `NullPointerException` it seems the receiver of the Status wouldn't gain too much by the added description. So although it would be nice, making the exception processing straight-forward seems more important.